homeassistant/helpers/update_coordinator.py

```diff
@@ -103,12 +104,13 @@ def __init__(
             randint(event.RANDOM_MICROSECOND_MIN, event.RANDOM_MICROSECOND_MAX) / 10**6
         )
 
-        self._listeners: dict[CALLBACK_TYPE, tuple[CALLBACK_TYPE, object | None]] = {}
+        self._listeners: dict[int, tuple[CALLBACK_TYPE, object | None]] = {}
         self._unsub_refresh: CALLBACK_TYPE | None = None
         self._unsub_shutdown: CALLBACK_TYPE | None = None
         self._request_refresh_task: asyncio.TimerHandle | None = None
         self.last_update_success = True
         self.last_exception: Exception | None = None
+        self._listener_id: int = 0
```

Comment:
nit: can we move this up, near `_listeners` just to keep related stuff together? and maybe rename to `_last_listener_id`